internal/component/prometheus/remote/queue/component.go

```diff
@@ -0,0 +1,160 @@
+package queue
+
+import (
+	"context"
+	"path/filepath"
+	"reflect"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/filequeue"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/network"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/serialization"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/types"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/storage"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "prometheus.remote.queue",
```

Comment:
Do you have plans to put other components in the `prometheus.remote` namespace?

Reply:
None, but I couldnt think of a better place to put it!